### PR TITLE
Added expand/collapse all button with icon to the Primary toolbar

### PR DESCRIPTION
**Pull request by Fewwy**
Described status expanded true/false in the state using useState
used property expandAll of PrimaryToolbar to pass it function collapseAll
image

### CODE REVIEW

File: src/Components/RecsListTable/RecsListTable.js [hunk 1]
```diff
@@ -497,6 +512,7 @@ const RecsListTable = ({ query }) => {
         />
       )}
       <PrimaryToolbar
+        expandAll={{ isAllExpanded, onClick: collapseAll }}
```
Comment (gkarat) [inline src/Components/RecsListTable/RecsListTable.js:515]: That is great that you used this simple property of the toolbar. However, I have spotted one bug. 

The `isAllExpanded` state parameter is not updated once you change the any of the filters/pagination. Imagine, that you have expanded all the rules, and if you navigate to the next page or apply any of the filters, you will see that the arrow button is still effective (= pointing downwards), but all the rows are hovered back.

I think, the only thing you need to enhance this `useEffect`: https://github.com/RedHatInsights/ocp-advisor-frontend/blob/c8fa96f39761ff0a05a93bfa0e3783a5a1ed2445/src/Components/RecsListTable/RecsListTable.js#L64

**Review by gkarat — CHANGES_REQUESTED** [review]

**Comment by gkarat** [comment]
Overall LGTM, but it seems we better treat this use case I have mentioned and then we can merge it.

**Comment by Fewwy** [comment]
@gkarat PTAL Should I add something else or is that's enough?

File: src/Components/RecsListTable/RecsListTable.js (hunk 1 above)
Comment (Fewwy) [reply src/Components/RecsListTable/RecsListTable.js:515]: Used terner operator in buildFilteredRows to have the same effect as you want
isOpen: isAllExpanded ? true : false

**Review by gkarat — APPROVED** [review]
Looks great, thank you, mergin'

File: src/Components/RecsListTable/RecsListTable.js [hunk 2]
```diff
@@ -83,7 +83,7 @@ const RecsListTable = ({ query }) => {
       .filter((rule) => passFilters(rule, filters))
       .map((value, key) => [
         {
-          isOpen: false,
+          isOpen: isAllExpanded ? true : false,
```
Comment (gkarat) [inline src/Components/RecsListTable/RecsListTable.js:86]:
```suggestion
          isOpen: isAllExpanded,
```

**Review by gkarat — APPROVED** [review]

**Pull request merged**